test/addons_test.go

```diff
@@ -265,13 +267,54 @@ func testgroup(t *testing.T, groupname string, version string, jobs ...clusterTe
 		return err
 	}
 
+	t.Logf("determining which addons in group %s need to be upgrade tested", groupname)
+	addonUpgrades := testharness.Loadables{}
+	for _, newAddon := range addons {
+		t.Logf("verifying whether upgrade testing is needed for addon %s", newAddon.GetName())
+		oldAddon, err := addontesters.GetLatestAddonRevisionFromLocalRepoBranch("../", comRepoRef, newAddon.GetName())
+		if err != nil {
+			return err
+		}
+		if oldAddon == nil {
+			t.Logf("no need to upgrade test %s, it appears to be a new addon (no previous revisions found in branch %s)", newAddon.GetName(), comRepoRef)
+			continue // new addon, upgrade test not needed
+		}
+
+		t.Logf("determining old and new versions for upgrade testing addon %s", newAddon.GetName())
+		oldVersion, err := semver.Parse(strings.TrimPrefix(oldAddon.GetAnnotations()[constants.AddonRevisionAnnotation], "v"))
+		if err != nil {
+			return err
+		}
+		newVersion, err := semver.Parse(strings.TrimPrefix(newAddon.GetAnnotations()[constants.AddonRevisionAnnotation], "v"))
+		if err != nil {
+			return err
+		}
+		if oldVersion.GT(newVersion) {
+			return fmt.Errorf("revisions for addon %s are broken, previous revision %s is newer than current %s", newAddon.GetName(), oldVersion, newVersion)
+		}
+		if !newVersion.GT(oldVersion) {
+			t.Logf("skipping upgrade test for addon %s, it has not be updated", newAddon.GetName())
+			continue
+		}
+
+		t.Logf("INFO: addon %s was modified and will be upgrade tested", newAddon.GetName())
```

Comment:
Same question https://github.com/mesosphere/kubeaddons/pull/979/files#r489378817 :D

Reply:
I followed up over there, let me know if that's satisfactory, of if you'd like to see something change!